ang/test/civicase/activity/services/activity-feed-measurements.service.spec.js

```diff
@@ -9,47 +9,94 @@
       ActivityFeedMeasurements = _ActivityFeedMeasurements_;
     }));
 
-    describe('getTopOffset()', function () {
-      var topOffset, expectedTopOffset;
+    describe('setScrollHeightOf()', function () {
+      var expectedTopOffset, elementToSetHeightTo;
+      var originaljQueryHeightFn = CRM.$.fn.height;
 
       beforeEach(function () {
-        addAdditionalMarkup();
-
-        var $feedPanelBody = CRM.$('.civicase__activity-feed>.panel-body');
-        var feedPanelBodyPaddingTop = parseInt($feedPanelBody.css('padding-top'));
-
-        topOffset = ActivityFeedMeasurements.getTopOffset();
-        expectedTopOffset =
-          CRM.$('.civicase__activity-feed__body').offset().top +
-          feedPanelBodyPaddingTop;
+        spyOn(CRM.$.fn, 'height').and.callThrough();
       });
 
       afterEach(function () {
         removeAdditionalMarkup();
+        CRM.$.fn.height = originaljQueryHeightFn;
+      });
+
+      describe('when used outside of contacts tab', function () {
+        beforeEach(function () {
+          addAdditionalMarkup();
+
+          CRM.$('.activity-feed-measurement-test-markup')
+            .append('<div class="element-to-set-height-to"></div>');
+          elementToSetHeightTo = CRM.$('.element-to-set-height-to');
+
+          var $feedPanelBody = CRM.$('.civicase__activity-feed>.panel-body');
+          var feedPanelBodyPaddingTop = parseInt($feedPanelBody.css('padding-top'));
+
+          ActivityFeedMeasurements.setScrollHeightOf(elementToSetHeightTo);
+          expectedTopOffset =
+            CRM.$('.civicase__activity-feed__body').offset().top +
+            feedPanelBodyPaddingTop;
+        });
+
+        it('sets the height of the sent element', function () {
+          // as the height is using 'calc', actual height could not be measured
+          expect(CRM.$.fn.height).toHaveBeenCalledWith('calc(100vh - ' + expectedTopOffset + 'px)');
+        });
       });
 
-      it('returns the distance of activity feed body from the top of the screen', function () {
-        expect(topOffset).toBe(expectedTopOffset);
+      describe('when used inside of contacts tab', function () {
+        beforeEach(function () {
+          addAdditionalMarkup(true);
```

Comment:
Instead of having this boolean value, why don't you append the `.crm-contact-tabs-list` mock markup here just as you are doing `.element-to-set-height-to` ?

Reply:
Fixed as discussed.